proposed/disable-multi-level-lookup-by-default.md

```diff
@@ -0,0 +1,222 @@
+    If the version requested by `global.json` is not found it will fail.
+* `dotnet` host commands
+    * Running `dotnet --info`, `dotnet --list-runtimes` and `dotnet --list-sdks`
+    with .NET 7+ `dotnet` will also disable multi-level lookup.
+    This may be considered a visible discrepancy. It will be possible to run
+    .NET 6 app via a private .NET 7 install `dotnet` through multi-level lookup,
+    but running `dotnet --list-runtimes` will not list such .NET 6 runtime as available.
+    * The output of `dotnet --info` will include a special message which will warn
+    about this discrepancy.
+    The output of `dotnet --info` is meant to be humanly readable text and
+    it already includes for example links to install more SDKs so adding
+    another message should have minimal risk of breaking anything.
+    `dotnet --list-runtimes` and `dotnet --list-sdks`
+    on the other hand are not formatted for humans and should not include
+    any additional message.
+* Native hosting APIs
+    * All hosting API scenarios have `.runtimeconfig.json` available and so will
+    act as applications - they will react to target framework version
+    as described above.
+    * `hostfxr_get_dotnet_environment_info` and `hostfxr_get_available_sdks`
+    would behave the same as `dotnet --list-runtimes` and `dotnet --list-sdks` -
+    so multi-level lookup is off on `hostfxr` .NET 7+.
+    * `hostfxr_resolve_sdk`, `hostfxr_resolve_sdk2` - similar to the enumeration
+    APIs these will not use multi-level lookup when running in `hostfxr` .NET 7+.
+    * `nethost`'s `get_hostfxr_path` API to find `hostfxr` does not react
+    to multi-level lookup in any way (effectively disabled always) and as such
+    is not affected by this change. This means that using private install location
+    will always run the `hostfxr` from the private install location, with no way
+    to allow multi-level lookup for `hostfxr` (existing behavior).
+
+## UX changes
+
+We want to inform the user about this change in some way in cases where it's
+permissible to do so. This means additional messages should not be printed when
+running applications since the app may depend on exact output format. As such
+the additional information can only be provided when running CLI commands (SDK).
+
+Host should print out a message when enumerating runtimes and SDKs in
+`dotnet --info`. As mentioned above `dotnet --list-runtimes` and `dotnet --list-sdks`
+must not be affected.
+
+The message should only be shown when there's observable difference in behavior.
+For CLI commands that means when running the command with multi-level lookup on
+would end up using different version of SDK than when running it with
+multi-level lookup off.
+
+`dotnet --info` should print the message in all cases where it's running
+from a private install location (so the global one doesn't match the current).
+
+We should look into UX around `dotnet run` and `dotnet test`, if these should
```

Comment:
in case of fx failure for 7+(or temporary) are we changing the UX or do we keep the standard one?